spring-integration-smb/src/main/java/org/springframework/integration/smb/outbound/SmbOutboundGateway.java

```diff
@@ -134,4 +187,88 @@ protected SmbFile enhanceNameWithSubDirectory(SmbFile file, String directory) {
 		}
 	}
 
+	@Override
+	protected List<?> ls(Message<?> message, Session<SmbFile> session, String dir) throws IOException {
+		return doTask(message, session, () -> super.ls(message, session, dir));
+	}
+
+	@Override
+	protected List<String> nlst(Message<?> message, Session<SmbFile> session, String dir) throws IOException {
+		return doTask(message, session, () -> super.nlst(message, session, dir));
+	}
+
+	@Override
+	protected File get(Message<?> message, Session<SmbFile> session, String remoteDir, String remoteFilePath,
+			String remoteFilename, SmbFile fileInfoParam) throws IOException {
+
+		return doTask(message, session,
+				() -> super.get(message, session, remoteDir, remoteFilePath, remoteFilename, fileInfoParam));
+	}
+
+	@Override
+	protected List<File> mGet(Message<?> message, Session<SmbFile> session, String remoteDirectory,
+			String remoteFilename) throws IOException {
+
+		return doTask(message, session,
+				() -> super.mGet(message, session, remoteDirectory, remoteFilename));
+	}
+
+	@Override
+	protected boolean rm(Message<?> message, Session<SmbFile> session, String remoteFilePath) throws IOException {
+		return doTask(message, session, () -> super.rm(message, session, remoteFilePath));
+	}
+
+	@Override
+	protected boolean mv(Message<?> message, Session<SmbFile> session, String remoteFilePath, String remoteFileNewPath)
+			throws IOException {
+
+		return doTask(message, session,
+				() -> super.mv(message, session, remoteFilePath, remoteFileNewPath));
+	}
+
+	@Override
+	protected String put(Message<?> message, Session<SmbFile> session, String subDirectory) {
+		try {
+			return doTask(message, session,
+					() -> super.put(message, session, subDirectory));
+		}
+		catch (IOException e) {
```

Comment:
Why is this? Doesn't look like `super.put()` throws any exceptions...

Reply:
It follows the same logic in FtpOutboundGateway